lib/mix/tasks/knigge/verify.ex

```diff
@@ -39,21 +39,52 @@ defmodule Mix.Tasks.Knigge.Verify do
 
   ## Options
 
-  At the moment `knigge.verify` offers no options. If you think this should be
-  different please open an issue.
+  --app (optional):
+    Name of the app for which the facades need to be verified.
+    Defaults to the current working environment app.
   """
 
+  @exit_codes %{unknown_app: 1, missing_modules: 2, unknown_options: 3}
+  @exit_reasons Map.keys(@exit_codes)
+
+  @unknown_error_code 64
+
   @impl Mix.Task
-  def run(_args) do
+  def run(raw_args) do
     Mix.Task.run("compile")
 
-    calling_app()
-    |> run_for()
-    |> exit_with()
+    {args, _argv, _errors} =
+
+    case OptionParser.parse(raw_args, strict: [app: :string]) do
+      {args, _argv, []} ->
+        app = case args[:app] do
+          nil -> calling_app()
+          app_string -> String.to_atom(app_string)
+        end
+
+        app
+        |> run_for()
+        |> exit_with()
```

Comment:
What's your take on simplifying this a bit and adding a `run_for/1` clause which transforms a binary into an atom?

```suggestion
      {opts, _argv, []} ->
        opts
        |> Keyword.get_lazy(:app, &calling_app/0)
        |> run_for()
        |> exit_with()
```

And `run_for/1` would then look like this:

```elixir
def run_for(app) when is_binary(app) do
  app
  |> String.to_atom()
  |> run_for()
end

def run_for(app) when is_atom(app) do
  # ...
end
```

As I see it this would read a lot clearer. 🙂

Reply:
TIL about `Keyword.get_lazy` :D

this does seem like a nice suggestion